### DISCUSSION TITLE
Терминология на FSH

### DISCUSSION THREAD

**Discussion by GorbachevaT** [fhir-ru/core#229] [General]
Коллеги, при описании NamingSystem в FSH ONLINE в первой же строке получаю ошибку. Слова "NamingSystem" белого цвета, система их "не знает": 
![1](https://github.com/fhir-ru/core/assets/127845097/c0ee475e-5275-4aef-bf19-bce95686fc2e)

При замене "NamingSystem :" на "CodeSystem :" ошибки нет, что я делаю не так?
![2](https://github.com/fhir-ru/core/assets/127845097/402b749b-300a-45d7-a238-2faa8db6acd2)

**Comment by eugenekogan-spb** [comment]
Можно отметить также, что в [руководстве fsh](https://build.fhir.org/ig/HL7/fhir-shorthand/reference.html) нет декларации NamingSystem:
<img width="1127" alt="image" src="https://github.com/fhir-ru/core/assets/62525329/a6239811-4e34-4cca-a18f-40c532d24da1">

**Reply by Vrodvi** [reply]
Не для всех типов ресурсов есть зарезервированные ключевые слова в FSH.
У ресурса NamingSystem специального ключевого слова нет, поэтому профиль для него нужно описывать как профиль простого ресурса (аналогично ресурсам Patient, Organization и аналогичным).
А вот обозначение одной конкретной системы идентификации - это будет уже не профиль, а экземпляр ресурса (аналогично тому как конкретный пациент Иванов Иван - это экземпляр ресурса Patient).

Поэтому для описания структуры систем идентификации в проекте Ru Core можно задать один общий профиль Core_NamingSystem (без каких-либо ограничений), например:
```
Profile:        Core_NamingSystem
Parent:              NamingSystem
Id:             core-namingsystem
Title:          "Core NamingSystem Система идентификации для Ru Core."
```

А сами системы идентификации описывать уже как экземпляры ресурса NamingSystem, соответствующие этому общему профилю, например для паспорта РФ:
```
Instance:   core-namingsystem-instance-core-ns-passport-rf
InstanceOf: Core_NamingSystem
Usage:      #definition

* status = #active
* name = "Пространство идентификаторов для паспортов РФ"
* kind = #identifier
* date = "2015-08-31"

* uniqueId[0].type  = #uri
* uniqueId[=].value = "http://fhir.ru/core/systems/passport-rf"
* uniqueId[=].preferred = true
```
